src/UnigramWordPredictor.java

```diff
@@ -99,9 +112,15 @@ public void train(Scanner scanner) {
    * @return the predicted next word, or null if no prediction can be made
    */
   public String predictNextWord(List<String> context) {
-    // TODO: Return a predicted word given the words preceding it
     // Hint: only the last word in context should be looked at
-    return null;
+    
+    String contextString = context.get(context.size()-1);
+    if(!neighborMap.containsKey(contextString)){
+      return null;
+    }
+    List<String> currentList = neighborMap.get(contextString);
+
+    return currentList.get(new Random().nextInt(currentList.size()));
```

Comment:
Great logic! You may consider splitting line 123 into multiple statements with intermediate variables. It does a lot in one line and may be a bit more readable if it was split up more. It is a matter of style and preference though!